PyDSS/pyPostprocessor/PostprocessScripts/EdLiFoControl.py

```diff
@@ -0,0 +1,567 @@
+                
+                if self.curtail_option == 'pctpmpp':
+                    self.oldpctpmpp = self.newpctpmpp
+                    if self.affected_buses and self.oldpctpmpp==0:
+                        self.comment=f"{pv_sys} is fully curtailed but overloads still exist!"
+                    else:
+                        pass
+                        
+                elif self.curtail_option=='kva':
+                    if self.affected_buses and max(self.pvs_df.loc[self.poa_lifo_ordered_pv_list,'kVARated'])==0:
+                        self.comment='All PV systems have been curtailed, but overloads still exist!'
+                    
+                
+                if not self.affected_buses:
+                    self.solution_status = 1
+                    self.comment = "Overload Solved!"
+                    self.logger.info(self.comment)
+                    
+                    break
+                
+            self.record_curtailment(pv_sys)        
+            
+        # Setting pmpps back to their original values    
+        if self.init_kVA!={}:
+            for pv_sys in self.my_lifo_list:
+                inipctpmpp = self.init_pctpmpp[pv_sys]
+                inikva = self.init_kVA[pv_sys]
+                dss.run_command(f"Edit PVSystem.{pv_sys} pctPmpp={inipctpmpp} kVA={inikva}")
+        
+        
+    def get_curtailment_candidates(self):
+        self.lifo_pv_list=[]
+        for pv in self.poa_lifo_ordered_pv_list:
+            if (self.user_lifo_pv_list is None) :
+                if self.pvdf3.loc[pv,'bus1'] in self.zone:
+                    self.lifo_pv_list.append(pv)
+            else:
+                dss.Circuit.SetActiveElement('PVSystem.'+str(pv))
+                bus1 = dss.Properties.Value("bus1")
+                if bus1 in self.zone:
+                    self.lifo_pv_list.append(pv)
+            
+            
+        self.lifo_pv_list = list(self.lifo_pv_list)
+        if self.zone and not self.lifo_pv_list:
+            self.comment = f'No PV system found in the vicinity (radius={self.zone_threshold}) of the thermal overload'
+            #print(comment)
+        
+        if self.zone_option=='Yes' and self.lifo_pv_list:
+            self.my_lifo_list = self.lifo_pv_list + [x for x in self.poa_lifo_ordered_pv_list if not x in self.lifo_pv_list]
+            
+        else:
+            self.my_lifo_list = self.poa_lifo_ordered_pv_list
+        
+    
+    def poa_curtail(self):
+        
+        sol = dss.Solution
+        dss.run_command("set stepsize=0")
+        dss.run_command("BatchEdit PVSystem.* VarFollowInverter=True") 
+        self.get_lifo_ordered_list()
+            
+        self.PV_curtailed = dict()
+        self.PV_kW_curtailed = dict()
+        self.pvbus_voltage = dict()
+        self.solution_status = -1
+        self.comment = 'RAS'
+        self.total_kVA_deployed = sum(self.pvs_df['kVARated'])

```

Comment:
It may not matter here but the pandas method will be much faster.
```
self.pvs_df['kVARated'].sum()
```

Reply:
Okay